### ISSUE TITLE
Backup

### ISSUE THREAD

**Issue by woodlogger** [snipe/snipe-it#2454]
#### Expected Behavior (or desired behavior if a feature request)

Click Generate a Backup and backup file is created.

---
#### Actual Behavior

When I click generate a backup I get a message that says:   Success: A new backup file was successfully created.
- It also says:  Backup files are located in: C:\Public\snipe-it\storage/app/backups

However, I see NO backup files in that location.  I have noticed that the slashes are flipped in the display.  I'm not sure if that has anything to do with it?

---
#### Please confirm you have done the following before posting your bug report:
- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](http://docs.snipeitapp.com/common-issues.html)

---
#### Please provide answers to these questions before posting your bug report:
- Snipe-It 3.3(fresh install)
- Windows Server 2012 R2
- install.sh method of install
- On page:  /admin/settings/backups

Laravel Log:  
[2016-08-17 13:28:43] production.ERROR: backups backups : error: Failed to backup because: The dump process failed with exitcode 1 : General error :

[2016-08-17 13:28:43] production.ERROR: backups backups : error: Failed to backup because: The dump process failed with exitcode 1 : General error :

**Comment by woodlogger** [comment]
Just looked at the Laravel.log file and I see:  
[2016-08-17 13:28:43] production.ERROR: backups backups : error: Failed to backup  because: The dump process failed with exitcode 1 : General error :   
[2016-08-17 13:28:43] production.ERROR: backups backups : error: Failed to backup  because: The dump process failed with exitcode 1 : General error :

**Comment by snipe** [comment]
Did you set the DB_DUMP_PATH in your env file to a valid path for mysqldump?
https://snipe-it.readme.io/docs/configuration#required-database-settings

**Comment by holyoaked** [comment]
Hi @snipe,

I'm not OP, but am having the same issue when I try to generate a backup.

I installed Snipe-IT using the install.sh script.

Would you be able to tell me where, exactly, the env file is? The only .env file I can find is docker.env, which has some, but not all, of the required settings.

I've also noticed that install.sh seems to have set my app environment to develop, not production. Am I able to change this to production without losing any data? Is the process to change envrionments simply switching the APP_ENV flag in docker.env?

**Comment by woodlogger** [comment]
Hey @snipe  I found another post about this at:  https://github.com/snipe/snipe-it/issues/2128 with a comment from BrettFagerlund it discusses escaping the path and granting access to the right directories.  This fixed my problem.  You may want to add this somewhere in the documentation.

Thanks

**Comment by avary69** [comment]
I had the same problem on Windows 2012 R2 with IIS 8.5. 

The issue from the previous posts aswell as the documenation is that the path might simply be wrong, depending on the installed version of MariaDB.

I used this command that converts the normal path to 8.3 formatted path that (strangely) is used here (just makes things harder than they need to be): `cmd /c for %A in ("C:\Program Files\MariaDB 10.1\bin") do @echo %~sA`. This reveals the the correct path for MariaDB 10.1 (actual version) is C:\\PROGRA~1\\MARIAD~1.1\\bin\\. 
C:\\PROGRA~1\\MARIAD~1.0\\bin\\ obviously was the path for version 10.0

@snipe: Please consider adding a hint to the documentation that the path is depending on the MariaDB version. It is easy to verify. Either use the above command or just copy that path (without the second backslashes) into Windows Explorer. It will give an error if it is the wrong path.

**Comment by sphere720** [comment]
@avary69 That did the trick for the artisan command!  The GUI version button still doesn't work but that's ok with me. I had a feeling the version number was not correct but had no idea the best way to figure out the correct one. The command script is genius. 
@snipe I second @avary69 in considering adding a note about using the script to determine the proper path for Windows servers. You know how MS likes to make things just a little more difficult that it needs to be. 
Thanks all for your assistance!

**Comment by avary69** [comment]
@sphere720 Other than other directories that need modify permissions for the GUI backup you also need to give IIS_IUSRS modify permissions (I did it on the \storage directory, though \backups should be enough). That should do the trick.

**Comment by sphere720** [comment]
@avary69 I do have the perms set for IIS_IUSRS to full on the \storage directory actually. There is definitely something wrong with my set up though. Because I am also unable to run any reports. Getting the permission denied error. Hoping I don't have to start from scratch.

**Comment by avary69** [comment]
@sphere720 what does the log say for the GUI backup?
Report issue: Did you set the permission for IUSR the other folders (\bootstrap\cache, public\upload, C:\Windows\Temp)?

**Comment by sphere720** [comment]
@avary69 After I replied, I went back to the instructions for IIS/Windows and noticed I had only set perms for IIS_IUSRS and not IUSR. Once I did that and tested both backup and reports in the GUI, it was all successful. Whew!  Thanks again for your guidance and to @snipe for the added IIS/Windows documentation. All is good and well for me know.

**Comment by avary69** [comment]
Happy to help.
@snipe Could you add the permissions also for the storage folder. That is missing for the Windows\IIS permissions documentation.
So both IUSR and IIS_IUSRS need modify permissions.

**Comment by sphere720** [comment]
Most triumphant!!!

**Comment by snipe** [comment]
So wait, do we still need the double-slashes? Other windows users reported that double slashes were necessary, like ``'C:\\PROGRA~1\\MARIAD~1.0\\bin'`

**Comment by snipe** [comment]
I've updated the docs here:

https://snipe-it.readme.io/v3.0/docs/windowsiis
https://snipe-it.readme.io/docs/configuration#required-database-settings

But not sure if that's exactly right. I took out the bit about htaccess, since according to #2490, since we have a web.config, we don't need to use the URL rewrite rules import anymore.

If anyone could take a moment to review for issues, I'd really appreciate it.

(Also if anyone running IIS felt like providing some screenshots to go with the text descriptions, I'd be forever grateful.)

**Comment by avary69** [comment]
@snipe For me both with and without ending double back slashes is working.

**Comment by JExactly** [comment]
I am Still having this issue. I have granted all appropriate permissions as listed above. I have escaped the slashes in the directory path, and I have used the shortpath to my bin folder for MySQL. 

My DB Dump path Setting: DB_DUMP_PATH="C:\\PROGRA~1\\MySQL\\MYSQLS~1.5\\bin"

Anyone have any additional Ideas on why it is not working for me? I am also running V3.3. I am thinking that maybe I missed something. But I have combed over the updated manuals, verified 4 times that I have done the above etc..

**Comment by avary69** [comment]
@JExactly Well, your setting does not show the escaped back slashes. Do you actually have them in your .env file?
Which version of MySQL do you use?

**Comment by sphere720** [comment]
@snipe I'm pulling some screenshots for you now. Also, I actually had to add not only IUSR (the one I was missing that was causing my backup GUI and reports GUI to error out with the perms message); but also had to add MYSERVERNAME\IIS_IUSRS. I gave it Full control. 

@JExactly, @avary69 is on to something. It was a huge part of my issue as well. Had the wrong version.

**Comment by JExactly** [comment]
@avary69 Yes, I actually didn't catch that I pasted the wrong string but it is verbatim to this in my ENV file: DB_DUMP_PATH="C:\\PROGRA~1\\MySQL\\MYSQLS~1.5\\bin\\"

My full path is C:\Program Files\MySQL\MySQL Server 5.5\bin

I have give IUSR and IIS_IUSRS full control to all folders..

**Comment by sphere720** [comment]
@snipe IIS Site Bindings Screenshot
![capture1](https://cloud.githubusercontent.com/assets/16905882/18476776/f8c16f48-7990-11e6-8b00-61a0574d6297.JPG)
Let me know what other screenshots you need. I'd be happy to provide.

**Comment by avary69** [comment]
@JExactly 2 things pop out
Still no double back slashes + double quotes, should be single. 
So, in your case should be this: DB_DUMP_PATH='C:\\PROGRA~1\\MySQL\\MYSQLS~1.5\\bin'
You can also verify the 8.3 path by copy/pasting it to the File Explorer (without the double back slashes)

**Comment by sphere720** [comment]
@JExactly Here is a screenshot of mine if that helps.
![capture](https://cloud.githubusercontent.com/assets/16905882/18477076/fd1dc2d4-7991-11e6-9ae7-7bb52ca42bad.JPG)

**Comment by JExactly** [comment]
I don't get it.. I retyped the exact identical path, and now it works.. It appears that there was an incorrect character, But I cannot identify where. AH well, Thank you both for the assistance as it is now working as it should.. Also as a note, I had used double quotes as well as single quotes. Single first with the correct path... I moved to double based on another post I had read.

**Comment by sphere720** [comment]
@JExactly Good deal. Sometimes you just gotta go back to the beginning of a section and start over. Thankfully, it was not an issue of having to completely start over. I will have to try it with the single quotes shortly to see if it still works and will share that here.

**Comment by avary69** [comment]
double quotes work for me too.

**Comment by sphere720** [comment]
Single quotes also work for me. So either will work.

**Comment by woodlogger** [comment]
Here is what I have for mine:  DB_DUMP_PATH='C:\\progra~1\\MySQL\\mysqls~1.7\\bin\\'

-Cheers

**Comment by crazycga** [comment]
Hey all.  Dunno if it's particularly important, but in my Windows 2012 R2 server environment, I had to make sure that BOTH IUSR and IIS_IUSRS had access to the C:\WINDOWS\TEMP directory.  Took me a bit to work out what was missing.  Hope this helps someone.

**Comment by sphere720** [comment]
Yes. I had mentioned that earlier in the thread I think.

Donald Counts
Exent Studio Productions Co.
Audio Engineer/Artist
405-527-5752

**Comment by Khelbun** [comment]
@crazycga Thanks a ton for that suggestion, I have the same setup of Server 2012 R2 and I just couldn't get this working until I added in IIS_IUSRS with modify access to the C:\windows\temp directory.

@sphere720 You had mentioned adding IIS_IUSRS to all the folders as well, I didn't need to do that I just needed to add them to the C:\windows\temp directory to get this working, less permissions is always better is possible :)

@snipe  This isn't currently in the documentation, and it took slogging through quite a few posts to find it, might be good to add it in.  Outside of that, thanks for all the great documentation you do have and all the help you provide on the forums, it's appreciated.
